pkg/kubectl/cmd/create_job.go

```diff
@@ -0,0 +1,163 @@
+
+	jobExample = templates.Examples(i18n.T(`
+		# Create a Job from a CronJob named "a-cronjob"
+		kubectl create job --from-cronjob=a-cronjob`))
+)
+
+type CreateJobOptions struct {
+	FromCronJob    string
+	FetchedCronJob *batchv1beta1.CronJob
+
+	OutputFormat  string
+	Namespace     string
+	V1Beta1Client clientbatchv1beta1.BatchV1beta1Interface
+	V1Client      clientbatchv1.BatchV1Interface
+	Mapper        meta.RESTMapper
+	Out           io.Writer
+	PrintObject   func(obj runtime.Object) error
+	PrintSuccess  func(mapper meta.RESTMapper, shortOutput bool, out io.Writer, resource, name string, dryRun bool, operation string)
+}
+
+// NewCmdCreateJob is a command to ease creating Jobs from CronJobs.
+func NewCmdCreateJob(f cmdutil.Factory, cmdOut io.Writer) *cobra.Command {
+	c := &CreateJobOptions{
+		Out: cmdOut,
+	}
+	cmd := &cobra.Command{
+		Use:     "job --from-cronjob=CRONJOB",
+		Short:   jobLong,
+		Long:    jobLong,
+		Example: jobExample,
+		Run: func(cmd *cobra.Command, args []string) {
+			cmdutil.CheckErr(c.Complete(f, cmd, args))
+			cmdutil.CheckErr(c.Validate())
+			cmdutil.CheckErr(c.RunCreateJob())
+		},
+	}
+	cmdutil.AddApplyAnnotationFlags(cmd)
+	cmdutil.AddValidateFlags(cmd)
+	cmdutil.AddPrinterFlags(cmd)
+	cmd.Flags().String("from-cronjob", "", "Specify the name of the CronJob to create a Job from.")
+
+	return cmd
+}
+
+func (c *CreateJobOptions) Complete(f cmdutil.Factory, cmd *cobra.Command, args []string) (err error) {
+	c.FromCronJob = cmdutil.GetFlagString(cmd, "from-cronjob")
+
+	// Complete other options for Run.
+	c.Mapper, _ = f.Object()
+
+	c.OutputFormat = cmdutil.GetFlagString(cmd, "output")
+
+	c.Namespace, _, err = f.DefaultNamespace()
+	if err != nil {
+		return err
+	}
+
+	c.PrintObject = func(obj runtime.Object) error {
+		return f.PrintObject(cmd, false, c.Mapper, obj, c.Out)
+	}
+
+	// need two client sets to deal with the differing versions of CronJobs and Jobs
+	clientset, err := f.KubernetesClientSet()
+	if err != nil {
+		return err
+	}
+	if c.V1Client == nil {
+		c.V1Client = clientset.BatchV1()
+	}
+	if c.V1Beta1Client == nil {
+		c.V1Beta1Client = clientset.BatchV1beta1()
+	}
+
+	fmt.Printf("%#v\n", c)
+
+	return nil
+}
+
+func (c *CreateJobOptions) Validate() error {
+	_, err := c.getCronJob(c.FromCronJob)
+	if err != nil {
+		return fmt.Errorf("failed to validate: %v", err)
+	}
+	return nil
+}
+
+func (c *CreateJobOptions) RunCreateJob() (err error) {
+	cronjob, err := c.getCronJob(c.FromCronJob)
+
+	if err != nil {
+		return fmt.Errorf("failed to fetch job: %v", err)
+	}
+
+	// fmt.Printf("DEBUG: CRONJOB RECEIVED: %#v\n", cronjob)
+
+	jobToCreate := &batchv1.Job{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:        cronjob.Name + "-manual-" + rand.String(3),
+			Namespace:   c.Namespace,
+			Annotations: make(map[string]string),
+		},
+		Spec: cronjob.Spec.JobTemplate.Spec,
+	}
+	jobToCreate.Annotations["cronjob.kubernetes.io/instantiate"] = "manual"
```

Comment:
IIRC we wanted to copy the labels from the template, similarly how the controller does it in [here](https://github.com/kubernetes/kubernetes/blob/89ac18de1a71d08d2bd7654c325cd8aa476190bd/pkg/controller/cronjob/utils.go#L177-L178).